### ISSUE TITLE
Extension topBarItems used before extension activation

### ISSUE THREAD

**Issue by panuhorsmalahti** [kontena/lens#5104]
![image](https://user-images.githubusercontent.com/2162413/160077353-cada45a7-2836-408f-8348-41faa54d1bcb.png)

This can cause a race condition where an exception such as "instance not created" is thrown.

Technical background:

This happens because during extension loading we set the extension instance to `instances` of `ExtensionLoader` before `onActivate` of the extension has been resolved. This causes `enabledExtensionInstances` `computed` to be updated, which causes `topBarItemsInjectable` to re-computed, which causes `topBarItems` of the extension be rendered. However, the topBarItems might use `IpcRenderer`, which is only created in `onActivate` of the extension (as is instructed in the official Extension documentation). If the `onActivate` is slow enough to call `IpcRenderer.createInstance`, the rendering will cause exception to be thrown.

Possible fix: set the extension to `instances` only after activation, or alternatively we could modify `enabledExtensionInstances` to only return activated extensions.

We could also consider to instruct extension authors to create IpcRenderer instance in `constructor`, but that wouldn't fix other race conditions (e.g. extension code might depend on the activation in other ways).

**Comment by stefcameron** [comment]
FYI, something in the 5.4.5 release fixed this issue (or the _actual_ issue, which may not have been this).

**Comment by jansav** [comment]
This was indeed fixed before. There's still possibility for this type of race-conditions but those are out-of-scope here. `ExtensionLoader` is the root problem here, it's doing things way to complicately.

Please inform if you still manage to reproduce the error.
